python/pyarrow/tensor.pxi

```diff
@@ -270,8 +279,10 @@ shape: {0.shape}""".format(self)
                                               &out_data, &out_coords))
         data = PyObject_to_object(out_data)
         coords = PyObject_to_object(out_coords)
-        result = coo_matrix((data[:, 0], (coords[:, 0], coords[:, 1])),
-                            shape=self.shape)
+        row, col = coords[:, 0], coords[:, 1]
+        result = coo_matrix((data[:, 0], (row, col)), shape=self.shape)
+        if self.has_canonical_format:
+            result.sum_duplicates()
```

Comment:
Why?

Reply:
@mrkn - wouldn't is_canonical guarantee no duplicates if indices were row- or column-ordered?

Reply:
@rok Yes, if `is_canonical` is true, the indices matrix is sorted in row-major order and doesn't have duplicates.
In other words, this flag has the same meaning as `has_canonical_format` of `scipy.sparse.coo_matrix` but the order is row-major, instead of column-major.

Reply:
So shouldn't this then be `if not self.has_canonical_format:`?

Reply:
No.  As I described in the above comment, the purpose of this `sum_duplicates` is converting the order from row-major to column-major.

Reply:
Wouldn't [sum_duplicates](https://github.com/scipy/scipy/blob/v1.5.1/scipy/sparse/coo.py#L526-L535) just terminate without doing anything in case `self.has_canonical_format`?
Sorry if I'm missing something here.

Reply:
Oh, we're checking if `SparseCOOTensor.has_canonical_format` not `scipy.coo_matrix.has_canonical_format`. Got it.
What about if the it's not canonical? Then we return noncanonical scipy object? Seems good.

Reply:
The `self` here is not an instance of `scipy.sparse.coo_matrix`, but it is a `SparseCOOTensor`.  So `self.has_canonical_format == True` means the index of the `SparseCOOTensor`, that is `(row, col)` pair, is sorted in row-major order.  Then `result.sum_duplicates` do sort `(row, col)` in column-major order.

Reply:
> What about if the it's not canonical? Then we return noncanonical scipy object? Seems good.

Yes, the canonicality is preserved.  The returned scipy object is also noncanonical when the original `SparseCOOTensor` is noncanonical.